File: 1779-find-nearest-point-that-has-the-same-x-or-y-coordinate/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.py
class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
        minDist = math.inf
        ans = -1
        for i in range(len(points)):
            if points[i][0]==x or points[i][1]==y: 
                if (abs(points[i][0]-x)+abs(points[i][1]-y))<minDist:
                    minDist = abs(points[i][0]-x)+abs(points[i][1]-y)
                    ans = i
        return ans

File: 1779-find-nearest-point-that-has-the-same-x-or-y-coordinate/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.c
int nearestValidPoint(int x, int y, int** points, int pointsSize, int* pointsColSize){
int index=-1,min=INT_MAX;
for(int i=pointsSize-1;i>=0;i--)
if ((points[i][0]==x)||(points[i][1]==y))
    if (abs(x-points[i][0])+abs(y-points[i][1])<= min){
                min = abs(points[i][0] - x) + abs(points[i][1] - y);
                index=i;
    }
return index;
}